buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -60,22 +60,23 @@
     private final String clusterName;
     private final NamedDomainObjectContainer<ElasticsearchNode> nodes;
     private final File workingDirBase;
-    private final File artifactsExtractDir;
+    private final Function<Integer, ElasticsearchDistribution> distributionFactory;
```

Comment:
I added a comment here previously which seems to have vanished. 

It seems that we are generating a distribution per node, defeating the purpose of `syncWithLinks`.
This optimization is important both in terms of disk space and time the build takes and I would prefer we don't merge this without it.  

It might make more sense to have this optimization in the distribution downloaded plugin trough and have test cluusters use the extracted distro dirs directly.

Reply:
The distribution download plugin already optimizes by only doing a single extraction for a given set of distribution properties. While we create a unique ElasticsearchDistribution object per node here, the underlying extracted dir will just exist once, in the root project.

Reply:
Thanks, got it, we only use the index because we need unique names for the domain object container.  
I wonder if we should be concerned about name clashes here ? In theory other uses of the distribution download plugin could use the same name as testclusters even trough unlikely.

Reply:
Clashes are possible but will result in an obvious error. I don't think we need to be too concerned about it.